runtime/vam/expr/arith.go

```diff
@@ -48,5 +52,74 @@ func (a *Arith) eval(vecs ...vector.Any) vector.Any {
 	if !ok {
 		return vector.NewStringError(a.zctx, coerce.ErrIncompatibleTypes.Error(), lhs.Len())
 	}
-	return f(lhs, rhs)
+	if a.opCode == vector.ArithDiv || a.opCode == vector.ArithMod {
+		defer func() {
+			if v := recover(); v != nil {
+				if err, ok := v.(runtime.Error); ok && err.Error() == "runtime error: integer divide by zero" {
+					out = a.evalDivideByZero(kind, lhs, rhs)
+				} else {
+					panic(v)
+				}
+			}
+		}()
+	}
+	out = f(lhs, rhs)
+	return
```

Comment:
Nit: Naked returns are the worst.
```suggestion
	return f(lhs, rhs)
```